tests/acceptance/MockReturn.feature

```diff
@@ -0,0 +1,78 @@
+Feature: MockReturn
+  
+  Background:
+    Given I have the following config
+      """
+      <?xml version="1.0"?>
+      <psalm totallyTyped="true">
+        <projectFiles>
+          <directory name="."/>
+          <ignoreFiles> <directory name="../../vendor"/> </ignoreFiles>
+        </projectFiles>
+        <plugins>
+          <pluginClass class="Psalm\MockeryPlugin\Plugin"/>
+        </plugins>
+      </psalm>
+      """
+    And I have the following code preamble
+      """
+      <?php
+      namespace NS;
+      use Mockery;
+      
+      """
+    
+  Scenario: Defined method mocking sets proper intersection return type (Psalm 3.0.9 and higher)
+    Given I have the following code
+      """
+      class User
+      {
+          /**
+           * @return void
+           */
+          public function someMethod()
+          {
+          
+          }
+      }
+      
+      $user = Mockery::mock('NS\User[someMethod]', []);
+      
+      if (is_array($user)) {
```

Comment:
This `is_array` check throws a psalm error, which is intentional so that the error message can be verified by this test (mainly that it contains the proper expected intersection type).

I tried other approaches here, but I couldn't get Psalm to throw an error when I removed the changes I made to the MockReturn hook.  I did confirm that without my changes, the intersection type isn't present here, as the error message changes to this:

>DocblockTypeContradiction | Cannot resolve types for $user - docblock-defined type Mockery\Mock does not contain array<array-key, mixed>